docs/DesignDocs/SGX_QuoteVerify_Integration.md

```diff
@@ -0,0 +1,525 @@
+#define OE_CLAIM_SGX_ROOT_CA_CRL_NUM "sgx_root_ca_crl_num"
+#define OE_CLAIM_SGX_TCB_EVAL_REF_NUM "sgx_tcb_eval_ref_num"
+#define OE_CLAIM_SGX_ROOT_KEY_ID "sgx_root_key_id"
+#define OE_CLAIM_SGX_PCK_PPID "sgx_pck_ppid"
+#define OE_CLAIM_SGX_TCB_CPUSVN "sgx_tcb_cpusvn"
+#define OE_CLAIM_SGX_TCB_PCE_ISVSVN "sgx_tcb_pce_isvsvn"
+#define OE_CLAIM_SGX_PCE_ID "sgx_pce_id"
+#define OE_CLAIM_SGX_TYPE "sgx_type"
+#define OE_CLAIM_SGX_PLATFORM_INSTANCE_ID "sgx_platform_instance_id"
+#define OE_CLAIM_SGX_DYNAMIC_PLATFORM "sgx_dynamic_platform"
+#define OE_CLAIM_SGX_CACHED_KEYS "sgx_cached_keys"
+#define OE_CLAIM_SGX_SMT_ENABLED "sgx_smt_enabled"
+
+#define OE_SGX_CLAIMS_COUNT 15
+```
+
+#### Options for retrieving SGX endorsements when using QVL-based quote verification
+There are two options for the OE SDK plugin library to retrieve SGX endorsements.
+
+##### Option 1: Keep existing implementation in SGX verifier plugin
+In the current plugin implementation, if the verifier doesn't provide SGX endorsements, it
+calls API `oe_get_sgx_endorsements()` to parse SGX ECDSA quote body to get PCK
+cert chain first, then calls OCALL API `oe_get_quote_verification_collateral_ocall()`
+to load Quote Provide Library(QPL) and connect to PCK Cert Caching Sever(PCCS)
+to get corresponding verification collaterals, including CRL, TCB and QE
+Identity, etc.
+
+##### Option 2: Ask DCAP QVL to retrieve endorsements
+If the verifier doesn't provide endorsements when calling DCAP QVL API, the QVL API
+implementation will parse quote body to get PCK cert chain, then load QPL
+and connect to PCCS to get corresponding verification
+collaterals.
+
+##### Proposal: Ask DCAP QVL to retrieve endorsements
+The proposal is to implement option 2, the reasons are:
+- The Intel SGX DCAP QVL already has logic to parse quote, and gets
+  verification collaterals
+- The collateral definition may change from time to time, e.g. DCAP 1.7 release
+  updated PCK cert extension to add new fields for support of multi-package
+  platforms
+- If we stay with the existing OE SDK implementation
+  - The implementation would need to be updated whenever the verification
+    collateral definition is changed
+  - The OE SDK logic for retrieving and parsing endorsements increases
+    verifier enclave TCB.
+
+With option 2, OE SDK doesn't need to maintain the complex logic for SGX collateral
+retrieving & parsing, nor need to be updated in response to changes
+in SGX collateral definition.
+
+
+#### Update existing plugin functions to only use DCAP QVL/QvE
+- In API `_verify_evidence()`, remove function call `oe_get_sgx_endorsements()`,
+  and only call `oe_parse_sgx_endorsements()` if caller provides an endorsement buffer
+- In API `_verify_report()` and relevant internal APIs, such as
+  `oe_verify_report_internal()` and `oe_verify_sgx_quote()`, remove the code that
+  loads QPL and retrieves endorsements
+- There is an API `__oe_sgx_set_minimum_crl_tcb_issue_date()`in header file
```

Comment:
Just FYI, double-underscore shouldn't be used as a prefix (and this is noted in the OE developer guidelines), so at some point that function should be renamed.

Reply:
> Just FYI, double-underscore shouldn't be used as a prefix (and this is noted in the OE developer guidelines), so at some point that function should be renamed.

Note that this is an existing API in current implementation